sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py

```diff
@@ -229,23 +244,28 @@ def receive(self, on_event, **kwargs):
          bool as the value indicating whether the starting_position for a specific partition is inclusive or not.
          This can also be a single bool value for all starting_position. The default value is False.
         :paramtype starting_position_inclusive: bool or dict[str,bool]
-        :keyword on_error: The callback function which would be called when there an error occurs during receiving.
+        :keyword on_error: The callback function which would be called when error is raised during receiving
+         after retry attempts exhaust and during the process of load balance.
          The callback takes two parameters: `partition_context` which contains partition information
-         and `error` being the exception. The callback should be defined like so: `on_error(partition_context, error)`.
+         and `error` being the exception. `partition_context` could be None if the error is raised during
+         the process of load balance. The callback should be defined like: `on_error(partition_context, error)`.
          The `on_error` callback will also be called if an unhandled exception is raised during
          the `on_event` callback.
         :paramtype on_error: Callable[[~azure.eventhub.PartitionContext, Exception]]
         :keyword on_partition_initialize: The callback function that will be called after a consumer for a certain
-         partition finishes initialization. The callback takes a single parameter: `partition_context`
+         partition finishes initialization. It would also be called when a new internal partition consumer is created
+         to take over the receiving process for a failed and closed internal partition consumer.
+         The callback takes a single parameter: `partition_context`
          which contains the partition information. The callback should be defined
-         like so: `on_partition_initialize(partition_context)`.
-        :paramtype on_partition_initialize: Callable[[~azure.eventhub.PartitionContext]]
+         like: `on_partition_initialize(partition_context)`.
+        :paramtype on_partition_initialize: Callable[[~azure.eventhub.aio.PartitionContext]]
```

Comment:
This is not the async consumer, so should point to sync docs.

Reply:
thanks, updated